File: servihogar-frontend/src/app/models/tecnico.ts
export interface TecnicoCategoriaRef {
  id_categoria: number;
  nombre: string;
}

export interface TecnicoZonaRef {
  id_zona: number;
  nombre: string;
}

export interface PortafolioItem {
  id_portafolio: number;
  titulo: string;
  descripcion: string | null;
  imagen_url: string;
  storage_path?: string | null;
}

export interface PortafolioCreateRequest {
  titulo: string;
  imagen_url: string;
  descripcion?: string;
}

export interface PortafolioItemPanel extends PortafolioItem {
  estado: string;
  fecha_subida: string;
}

export interface Tecnico {
  id_tecnico: number;
  nombres: string;
  apellidos: string;
  descripcion: string | null;
  experiencia_anios: number;
  calificacion: number | null;
  categorias: TecnicoCategoriaRef[];
  zonas: TecnicoZonaRef[];
}

export interface TecnicoDetalle extends Tecnico {
  portafolio: PortafolioItem[];
}

export type TecnicoPerfil = TecnicoDetalle;

File: servihogar-backend/backend/src/schemas/tecnico.py
from datetime import datetime

from pydantic import BaseModel, Field


class TecnicoCategoriaRef(BaseModel):
    id_categoria: int
    nombre: str


class TecnicoZonaRef(BaseModel):
    id_zona: int
    nombre: str


class PortafolioItem(BaseModel):
    id_portafolio: int
    titulo: str
    descripcion: str | None = None
    imagen_url: str
    storage_path: str | None = None


class PortafolioCreateRequest(BaseModel):
    titulo: str = Field(max_length=150)
    imagen_url: str
    descripcion: str | None = None


class PortafolioItemResponse(BaseModel):
    id_portafolio: int
    titulo: str
    descripcion: str | None = None
    imagen_url: str
    storage_path: str | None = None
    estado: str
    fecha_subida: datetime


class TecnicoResponse(BaseModel):
    id_tecnico: int
    nombres: str
    apellidos: str
    descripcion: str | None = None
    experiencia_anios: int
    calificacion: float | None = None
    categorias: list[TecnicoCategoriaRef] = []
    zonas: list[TecnicoZonaRef] = []


class TecnicoDetalleResponse(TecnicoResponse):
    portafolio: list[PortafolioItem] = []
